pkg/build/gobuild.go

```diff
@@ -735,6 +747,19 @@ func (g *gobuild) Build(ctx context.Context, s string) (Result, error) {
 		return nil, err
 	}
 
+	// Annotate the base image we pass to the build function with
+	// annotations indicating the digest (and possibly tag) of the
+	// base image.  This will be inherited by the image produced.
+	if mt != types.DockerManifestList {
```

Comment:
I moved this up because it gets the job done and it avoids me needing to create an `ocimutate.Annotations` so this preserves the SBOM attachment.